service/controller/app/v1/resource/chart/resource.go

```diff
@@ -32,19 +39,32 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 	return nil, nil
 }
 
-func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
-	return nil, nil
-}
-
 func (r *Resource) NewUpdatePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*controller.Patch, error) {
-	return nil, nil
+	patch := controller.NewPatch()
+	patch.SetCreateChange(desiredState)
+
+	return patch, nil
 }
 
 func (r *Resource) NewDeletePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*controller.Patch, error) {
 	return nil, nil
 }
 
 func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange interface{}) error {
+	app, err := key.ToCustomResource(obj)
```

Comment:
Same here. This can be added later this PR should just implement the desired state.